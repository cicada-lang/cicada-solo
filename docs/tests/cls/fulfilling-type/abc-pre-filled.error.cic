function the_nat_type(_: Trivial): Type {
  return Nat
}

class ABC {
  a: Trivial = sole
  b: the_nat_type(a)
  c: String
}

abc: ABC(sole, 2) = {
  a: sole
  b: 2
  c: "c"
}

abc

abc.a
abc.b
abc.c


class ABCDE {
  a: Trivial
  b: the_nat_type(a)
  c: String = "c"
  d: String
  e: String
}

abcde: ABCDE(sole, 2, "d") = {
  a: sole
  b: 2
  c: "c" // NOTE "c" is not equal to "d"
  d: "d"
  e: "e"
}

abcde

abcde.a
abcde.b
abcde.c
abcde.d
abcde.e
