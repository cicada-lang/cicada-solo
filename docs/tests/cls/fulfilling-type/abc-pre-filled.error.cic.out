I found syntax error in file: abc-pre-filled.error.cic

 10 |
 11 |let abc: ABC(sole, 2) = {
 12 |  a: sole
 13 |  b: 2
 14 |  c: "c"
 15 |}
Found the identifier "b" in [164, 165], while expecting:
 ",":
     zero_or_more:more@42 -> (head: property):field@45 > "," (tail: zero_or_more) 
 ",":
     optional:one@45 -> > "," 
 "}":
     operand:obj@41 -> "{"@42 (properties: zero_or_more):zero@42 (last_property: property):field@45 optional:zero@45 > "}" 
 and so on ...